### PR TITLE
Adds multiprocessing notes to the docs [ch733]

### PR DESCRIPTION
**Pull request by bbengfort**
Created a page for notes on multiprocessing with BTrDB -- it's a bit rough, but it took me a while to figure out this bug, so it'll be useful to remember in the future. 

Additionally, I added a Makefile for the docs as well as some requirements.txt to help folks build the documentation correctly.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -70,6 +70,7 @@ instance/
 .scrapy
 
 # Sphinx documentation
+docs/build/
```
Comment (looselycoupled) [inline .gitignore]: This is odd to me as git was already ignoring it on my laptop.  I had assumed the folder was already in there.

File: docs/Makefile [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+pseudoxml:
+	$(SPHINXBUILD) -b pseudoxml $(ALLSPHINXOPTS) $(BUILDDIR)/pseudoxml
+	@echo
+	@echo "Build finished. The pseudo-XML files are in $(BUILDDIR)/pseudoxml."
+
+.PHONY: dummy
+dummy:
+	$(SPHINXBUILD) -b dummy $(ALLSPHINXOPTS) $(BUILDDIR)/dummy
+	@echo
+	@echo "Build finished. Dummy builder generates no files."
```
Comment (looselycoupled) [inline docs/Makefile]: II assume you want to leave the html target in the root Makefile for convenience?

File: docs/requirements.txt [hunk 3]
```diff
@@ -0,0 +1,2 @@
+alabaster>=0.7.12
```
Comment (looselycoupled) [inline docs/requirements.txt]: Alabaster is a default Sphinx theme and is included with sphinx.  I'm curious why you want to include an explicit reference to it.

File: .gitignore (hunk 1 above)
Comment (bbengfort) [reply .gitignore]: It was ignoring `_build` (though there might be a generic `build`, e.g. not `docs/build`) - which is standard for Sphinx, but I changed it to `build` to match the `source` directory under `docs`.

File: docs/Makefile (hunk 2 above)
Comment (bbengfort) [reply docs/Makefile]: Ah, I didn't see that target in the root Makefile; I wasn't sure how you were building the docs - now I know! I'll put a note in the README about building the docs for future reference. I suppose we can get rid of this Makefile then too.

File: docs/requirements.txt (hunk 3 above)
Comment (bbengfort) [reply docs/requirements.txt]: The version of alabaster that I had with Sphinx 1.7 was issuing errors, upgrading to this version of alabaster fixed the errors (note that even with the errors it still built). Potentially upgrading to Sphinx 1.8 may have also fixed the errors, but figured it was better to be clear which version was required.

**Pull request merged**